gadgets/mri_core/GenericReconBase.cpp

```diff
@@ -22,6 +23,18 @@ namespace Gadgetron {
         {
             Gadgetron::get_debug_folder_path(debug_folder.value(), debug_folder_full_path_);
             GDEBUG_CONDITION_STREAM(verbose.value(), "Debug folder is " << debug_folder_full_path_);
+
+            // Create debug folder if necessary
+            boost::filesystem::path boost_folder_path(debug_folder_full_path_);
+            try
```

Comment:
I'm not sure that this is necessary.  According to boost's documentation, the function create_directories() "establishes the postcondition by calling create_directory() for any element of p that does not exist."  I interpreted this to mean that it would only call create_directory() for non-existing folders.  I think this is also backed up by line 966 here, if I'm interpreting that correctly as well: https://github.com/boostorg/filesystem/blob/develop/src/operations.cpp

Also, the documentation for create_directory() also says "Creation failure because p resolves to an existing directory shall not be treated as an error."

I'm certainly no expert at boost though, so I'm happy to add an is_directory() call if this is appropriate.